align/compiler/create_array_hierarchy.py

```diff
@@ -10,6 +10,7 @@
 from .merge_nodes import merge_nodes
 from .util import compare_two_nodes
 from ..schema import constraint
+from ..schema.hacks import HierDictNode
```

Comment:
Why calling this hacks?

Reply:
There are a couple of reasons I call these data structures hacks:
1. These are temporary data structures created exclusively to splice constraints into the existing auto-annotation flow
2. They emulate dictionaries by overriding __setitem__, __getitem__ and __contains__ but they are not dicts
   (Some dict operations may result in undefined behavior)
3. They allow extra attributes to be added at any time making the type checking very lax
4. They are somewhat redundant in the information they represent and do not provide a shared representation to go back and forth between engines. 
   (HierDictNode & VerilogJson* both represent hierarchy information. We should be trying to replace it with a single shared data representation using SubCircuit, Instance etc)

Reply:
HierDictNode is an internal data structure used by auto-annotation. The top-level dictionary (Dict[str, HierDictNode]) is called self.hier_graph_dict hence the data structure name.